lib/models/price_wrapper.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'price_wrapper.freezed.dart';
+part 'price_wrapper.g.dart';
+
+@freezed
+
+/// Contains all the details associated with a Price
+class Price with _$Price {
+  const factory Price(
+    /// Formatted price of the item, including its currency sign. For example $3.00
+    @JsonKey(name: 'formatted') String formatted,
+
+    /// Price in micro-units, where 1,000,000 micro-units equal one unit of the currency.
+    ///
+    /// For example, if price is "€7.99", price_amount_micros is 7,990,000. This value represents
+    /// the localized, rounded price for a particular currency.
+    @JsonKey(name: 'amountMicros') int amountMicros,
+
+    /// Returns ISO 4217 currency code for price and original price.
```

Comment:
isn't it 8601?

Reply:
That's for time. 4217 is for currency codes

Reply:
thank you, TIL. was trying to see how 8601 could be typoed to 4217 and it didn't seem to make sense haha